pkg/controller.v1beta1/experiment/manifest/generator_test.go

```diff
@@ -245,18 +237,17 @@ spec:
 
 	expectedRunSpec, err := util.ConvertStringToUnstructured(expectedStr)
 	if err != nil {
```

Comment:
`expectedRunSpec` is not evaluated using a function in `generator.go` so I am not sure if the errors defined in `generator.go` are applicable here.

Reply:
I think we might need to define a dedicated error since the `YAMLSyntaxError` struct's field `err` cannot be exported and we therefore cannot initialize a `YAMLSyntaxError` struct in `generator_test.go` to define the expected error.

Reply:
I think it is better to move both `errConvertStringToUnstructuredFailed` and `errConvertUnstructuredToStringFailed` to [katib/pkg/controller.v1beta1/util/unstructured.go](https://github.com/kubeflow/katib/blob/7df05c23a55363ba7cbbd7b9859943562afd71f4/pkg/controller.v1beta1/util/unstructured.go).

Please let me know your thoughts.

Reply:
@tariq-hasan I was missing any context here. Sorry for your confusion.
I meant like this.

```golang
	if diff := cmp.Diff(nil, err, cmpopts.EquateErrors()); len(diff) != 0 {
		t.Errorf("ConvertStringToUnstructured failed (-want,+got):\n%s", diff)
	}
```

Reply:
I addressed the comment in this [commit](https://github.com/kubeflow/katib/pull/2289/commits/e8afe1466024ac6e216736c391c8d0f782fae4fe).